infra/modules/azure_function_app/function_app.tf

```diff
@@ -0,0 +1,87 @@
+resource "azurerm_service_plan" "this" {
+  name                   = "${local.project}-${var.domain}-${var.app_name}-asp-${var.instance_number}"
+  location               = var.location
+  resource_group_name    = var.resource_group_name
+  os_type                = "Linux"
+  sku_name               = local.function_app.sku_name
+  zone_balancing_enabled = local.function_app.zone_balancing_enabled
+
+  tags = var.tags
+}
+
+resource "azurerm_linux_function_app" "this" {
+  name                = "${local.project}-${var.domain}-${var.app_name}-func-${var.instance_number}"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+
+  service_plan_id = azurerm_service_plan.this.id
```

Comment:
we should enable module consumers to optionally specify a different `service_plan_id` to allow `asp` reuse

Reply:
(should we split `asp` module and `function_app` module?)

Reply:
Changes done